jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java

```diff
@@ -381,26 +385,40 @@ public Interceptor getNextInterceptor() {
     VmApiProxyEnvironment getRequestSpecificEnvironment() {
       return requestSpecificEnvironment;
     }
+    
+    @Override
+    public String toString()
+    {
+      return String.format("RequestContext@%x %s==%s",hashCode(),request.getRequestURI(),requestSpecificEnvironment);
+    }
   }
-  
+
   public class ContextListener implements ContextHandler.ContextScopeListener, ServletRequestListener {
     @Override
     public void enterScope(org.eclipse.jetty.server.handler.ContextHandler.Context context, Request baseRequest, Object reason) {
       RequestContext requestContext = getRequestContext(baseRequest);
-      VmApiProxyEnvironment environment=requestContext.getRequestSpecificEnvironment();
-      String traceId=environment.getTraceId();
-      if (traceId!=null)
-        LogContext.current().put("traceId", environment.getTraceId());
-      if (LOG.isDebugEnabled())
-        LOG.debug("Enter {} -> {}",ApiProxy.getCurrentEnvironment(),environment);
-      ApiProxy.setEnvironmentForCurrentThread(environment); 
+      if (requestContext==null) {
+        if (logger.isLoggable(Level.FINE))
+          logger.fine("Entered with default env");
```

Comment:
braces. 

Also, is it worth doing the check when you are using one constant string?